### PR TITLE
docs(agw): Adding steps to build containerized AGW on Aarch64

### PR DESCRIPTION
**Pull request by m-govind**
Signed-off-by: Govindarajan <govindarajan.mohandoss@arm.com>

## Additional Information

- [ ] This change is backwards-breaking

### CODE REVIEW

**Comment by maxhbr** [comment]
Just tried to run the command on a **non-arm machine**, and it fails with the following log:

```
$ docker-compose build --build-arg CPU_ARCH=aarch64 --build-arg DEB_PORT=arm64
...
 => [agw_gateway_python internal] load metadata for docker.io/library/ubuntu:focal                                                                                                                         154.8s
 => [agw_gateway_c internal] load build context                                                                                                                                                              0.3s
 => => transferring context: 13.97MB                                                                                                                                                                         0.3s
 => CACHED [agw_gateway_python gateway_c  1/50] FROM docker.io/library/ubuntu:focal                                                                                                                          0.0s
 => [agw_gateway_python internal] load build context                                                                                                                                                         1.2s
 => => transferring context: 59.82MB                                                                                                                                                                         1.1s
 => [agw_gateway_c gateway_c  2/50] RUN apt-get update && apt-get install -y   apt-utils   apt-transport-https   ca-certificates   wget   gnupg   sudo   netcat   libyaml-cpp-dev   libgoogle-glog-dev   l  47.3s
 => [agw_gateway_c builder  2/25] RUN apt-get update &&   apt-get install -y apt-utils software-properties-common apt-transport-https gnupg wget &&   wget -P /usr/sbin --progress=dot:giga https://github  49.7s
 => [agw_gateway_python gateway_python  2/20] RUN apt-get update && apt-get install -y   apt-transport-https   ca-certificates   netcat   sudo   python3-pip   python3-venv   virtualenv   python3-eventle  78.2s
 => [agw_gateway_python builder  2/10] RUN apt-get update && apt-get install -y   wget   ruby   sudo   ruby-dev   docker.io   python3-pip   python3-dev   python3-eventlet   python3-pystemd   python3-pr  153.1s
 => [agw_gateway_c gateway_c  3/50] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                              0.3s
 => [agw_gateway_c gateway_c  4/50] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                                      0.6s
 => [agw_gateway_c gateway_c  5/50] RUN apt-get update && apt-get install -y   libopenvswitch   openvswitch-datapath-dkms   openvswitch-common   openvswitch-switch                                         38.7s
 => [agw_gateway_c builder  3/25] RUN apt-get update && apt-get install -y   sudo   curl   wget   unzip   cmake   git   build-essential   autoconf   libtool   pkg-config   libgflags-dev   libc++-dev   p  87.4s
 => [agw_gateway_python gateway_python  3/20] RUN cd /usr/local/bin && ln -s /usr/bin/python3 pytho &&   python3 -m venv /build                                                                              2.3s
 => [agw_gateway_python gateway_python  4/20] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                    0.6s
 => [agw_gateway_python gateway_python  5/20] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                            0.8s
 => [agw_gateway_python gateway_python  6/20] RUN echo "deb https://packages.fluentbit.io/ubuntu/focal focal main" > /etc/apt/sources.list.d/tda.list                                                        0.5s
 => [agw_gateway_python gateway_python  7/20] RUN wget -qO - https://packages.fluentbit.io/fluentbit.key | apt-key add -                                                                                     1.2s
 => [agw_gateway_python gateway_python  8/20] RUN apt-get update && apt-get install -y   td-agent-bit   libopenvswitch   openvswitch-datapath-dkms   openvswitch-common   openvswitch-switch   bcc-tools    25.3s
 => [agw_gateway_c builder  4/25] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                                0.5s
 => [agw_gateway_c builder  5/25] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                                        0.9s
 => [agw_gateway_c builder  6/25] RUN apt-get update && apt-get install -y   libfolly-dev   oai-asn1c   oai-freediameter   oai-gnutls   oai-nettle   magma-cpp-redis   magma-libfluid   grpc-dev   prometh  14.1s
 => [agw_gateway_c builder  7/25] RUN rm /etc/apt/sources.list.d/magma.list                                                                                                                                  0.4s
 => [agw_gateway_c builder  8/25] WORKDIR /magma                                                                                                                                                             0.3s
 => [agw_gateway_python builder  3/10] RUN cd /usr/local/bin && ln -s /usr/bin/python3 python                                                                                                                0.6s
 => [agw_gateway_c builder  9/25] COPY WORKSPACE.bazel BUILD.bazel .bazelignore .bazelrc .bazelversion /magma/                                                                                               0.2s
 => [agw_gateway_c builder 10/25] COPY bazel/ /magma/bazel                                                                                                                                                   0.2s
 => CANCELED [agw_gateway_python builder  4/10] RUN gem install fpm                                                                                                                                          1.0s
 => ERROR [agw_gateway_c builder 11/25] RUN bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_jso  0.5s
------
 > [agw_gateway_c builder 11/25] RUN bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_json//:json   @sentry_native//:sentry:
#0 0.383 /bin/sh: 1: bazel: not found
------
failed to solve: executor failed running [/bin/sh -c bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_json//:json   @sentry_native//:sentry]: exit code: 127
```
Without the `build-args` it works.

Is it expected to not work on non-arm machines?

**Comment by m-govind** [comment]
> Just tried to run the command on a **non-arm machine**, and it fails with the following log:
> 
> ```
> $ docker-compose build --build-arg CPU_ARCH=aarch64 --build-arg DEB_PORT=arm64
> ...
>  => [agw_gateway_python internal] load metadata for docker.io/library/ubuntu:focal                                                                                                                         154.8s
>  => [agw_gateway_c internal] load build context                                                                                                                                                              0.3s
>  => => transferring context: 13.97MB                                                                                                                                                                         0.3s
>  => CACHED [agw_gateway_python gateway_c  1/50] FROM docker.io/library/ubuntu:focal                                                                                                                          0.0s
>  => [agw_gateway_python internal] load build context                                                                                                                                                         1.2s
>  => => transferring context: 59.82MB                                                                                                                                                                         1.1s
>  => [agw_gateway_c gateway_c  2/50] RUN apt-get update && apt-get install -y   apt-utils   apt-transport-https   ca-certificates   wget   gnupg   sudo   netcat   libyaml-cpp-dev   libgoogle-glog-dev   l  47.3s
>  => [agw_gateway_c builder  2/25] RUN apt-get update &&   apt-get install -y apt-utils software-properties-common apt-transport-https gnupg wget &&   wget -P /usr/sbin --progress=dot:giga https://github  49.7s
>  => [agw_gateway_python gateway_python  2/20] RUN apt-get update && apt-get install -y   apt-transport-https   ca-certificates   netcat   sudo   python3-pip   python3-venv   virtualenv   python3-eventle  78.2s
>  => [agw_gateway_python builder  2/10] RUN apt-get update && apt-get install -y   wget   ruby   sudo   ruby-dev   docker.io   python3-pip   python3-dev   python3-eventlet   python3-pystemd   python3-pr  153.1s
>  => [agw_gateway_c gateway_c  3/50] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                              0.3s
>  => [agw_gateway_c gateway_c  4/50] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                                      0.6s
>  => [agw_gateway_c gateway_c  5/50] RUN apt-get update && apt-get install -y   libopenvswitch   openvswitch-datapath-dkms   openvswitch-common   openvswitch-switch                                         38.7s
>  => [agw_gateway_c builder  3/25] RUN apt-get update && apt-get install -y   sudo   curl   wget   unzip   cmake   git   build-essential   autoconf   libtool   pkg-config   libgflags-dev   libc++-dev   p  87.4s
>  => [agw_gateway_python gateway_python  3/20] RUN cd /usr/local/bin && ln -s /usr/bin/python3 pytho &&   python3 -m venv /build                                                                              2.3s
>  => [agw_gateway_python gateway_python  4/20] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                    0.6s
>  => [agw_gateway_python gateway_python  5/20] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                            0.8s
>  => [agw_gateway_python gateway_python  6/20] RUN echo "deb https://packages.fluentbit.io/ubuntu/focal focal main" > /etc/apt/sources.list.d/tda.list                                                        0.5s
>  => [agw_gateway_python gateway_python  7/20] RUN wget -qO - https://packages.fluentbit.io/fluentbit.key | apt-key add -                                                                                     1.2s
>  => [agw_gateway_python gateway_python  8/20] RUN apt-get update && apt-get install -y   td-agent-bit   libopenvswitch   openvswitch-datapath-dkms   openvswitch-common   openvswitch-switch   bcc-tools    25.3s
>  => [agw_gateway_c builder  4/25] RUN echo "deb https://artifactory.magmacore.org/artifactory/debian-test focal-ci main" > /etc/apt/sources.list.d/magma.list                                                0.5s
>  => [agw_gateway_c builder  5/25] RUN wget -qO - https://artifactory.magmacore.org:443/artifactory/api/gpg/key/public | apt-key add -                                                                        0.9s
>  => [agw_gateway_c builder  6/25] RUN apt-get update && apt-get install -y   libfolly-dev   oai-asn1c   oai-freediameter   oai-gnutls   oai-nettle   magma-cpp-redis   magma-libfluid   grpc-dev   prometh  14.1s
>  => [agw_gateway_c builder  7/25] RUN rm /etc/apt/sources.list.d/magma.list                                                                                                                                  0.4s
>  => [agw_gateway_c builder  8/25] WORKDIR /magma                                                                                                                                                             0.3s
>  => [agw_gateway_python builder  3/10] RUN cd /usr/local/bin && ln -s /usr/bin/python3 python                                                                                                                0.6s
>  => [agw_gateway_c builder  9/25] COPY WORKSPACE.bazel BUILD.bazel .bazelignore .bazelrc .bazelversion /magma/                                                                                               0.2s
>  => [agw_gateway_c builder 10/25] COPY bazel/ /magma/bazel                                                                                                                                                   0.2s
>  => CANCELED [agw_gateway_python builder  4/10] RUN gem install fpm                                                                                                                                          1.0s
>  => ERROR [agw_gateway_c builder 11/25] RUN bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_jso  0.5s
> ------
>  > [agw_gateway_c builder 11/25] RUN bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_json//:json   @sentry_native//:sentry:
> #0 0.383 /bin/sh: 1: bazel: not found
> ------
> failed to solve: executor failed running [/bin/sh -c bazel build   @com_github_grpc_grpc//:grpc++   @com_google_protobuf//:protobuf   @prometheus_cpp//:prometheus-cpp   @yaml-cpp//:yaml-cpp   @github_nlohmann_json//:json   @sentry_native//:sentry]: exit code: 127
> ```
> 
> Without the `build-args` it works.
> 
> Is it expected to not work on non-arm machines?

This is the command to build containerized AGW on Arm platform with 5.4 kernel. Did you try this command on Arm with 5.4 kernel ?

File: lte/gateway/docker/README.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ containerized AGW on AWS](#deploying-the-containerized-agw-on-aws).
 ## Building the images
 
 The images can be built with `cd $MAGMA_ROOT/lte/gateway/docker && docker-compose build`.
+On Arm architecture with 5.4 kernel:The images can be built with `cd $MAGMA_ROOT/lte/gateway/docker && docker-compose build --build-arg CPU_ARCH=aarch64 --build-arg DEB_PORT=arm64`.
```
Comment (voisey) [inline lte/gateway/docker/README.md:17]:
```suggestion
On an Arm architecture with the 5.4 kernel: The images can be built with `cd $MAGMA_ROOT/lte/gateway/docker && docker-compose build --build-arg CPU_ARCH=aarch64 --build-arg DEB_PORT=arm64`.
```

**Comment by maxhbr** [comment]
> This is the command to build containerized AGW on Arm platform with 5.4 kernel. Did you try this command on Arm with 5.4 kernel ?

I ran it on x86. But the error confuses me. Why is the different architecture causing a `/bin/sh: 1: bazel: not found` in a docker build?

Can you tell me how to test it e.g. withing qemu?

**Comment by m-govind** [comment]
> > This is the command to build containerized AGW on Arm platform with 5.4 kernel. Did you try this command on Arm with 5.4 kernel ?
> 
> I ran it on x86. But the error confuses me. Why is the different architecture causing a `/bin/sh: 1: bazel: not found` in a docker build?
> 
> Can you tell me how to test it e.g. withing qemu?

What is the command that you run on x86 and on Arm ? Nils Semmelrock have already built the image on Arm.

**Review by maxhbr — APPROVED** [review]
LGTM

**Comment by maxhbr** [comment]
Was able to use these commands to build on a RPi 4 4G :+1:

**Comment by maxhbr** [comment]
can you rebase, please?

**Comment by m-govind** [comment]
> can you rebase, please?

Now it looks good except codecov part. Please merge this change.

**Pull request merged**